### PR TITLE
Add pytest

### PR DESCRIPTION
**Pull request by joeyballentine**
Doesn't add any useful tests yet, just adds the testing framework

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
+## JS
+
```
Comment (RunDevelopment) [inline .gitignore:2]: ?

Comment (joeyballentine) [reply .gitignore:2]: I combined the root gitignores and the backend gitignores, so I separated them out

Comment (joeyballentine) [reply .gitignore:2]: perhaps JS wasn't the best section name to give that

**Comment by RunDevelopment** [comment]
Why did you use `tests/backend` for backend tests? Could you use `backend/tests` instead? That would also mean that you could remove a `__init__.py`s.

**Comment by joeyballentine** [comment]
> Why did you use `tests/backend` for backend tests? Could you use `backend/tests` instead? That would also mean that you could remove a `__init__.py`s.

I could yes, but then I would be bundling the tests with chaiNNer, unless I find a way to specifically ignore a folder from the extraResources electron option

**Comment by RunDevelopment** [comment]
> unless I find a way to specifically ignore a folder from the extraResources electron option

[Seems easy enough](https://www.electron.build/file-patterns#multiple-glob-patterns).

**Comment by joeyballentine** [comment]
> > unless I find a way to specifically ignore a folder from the extraResources electron option
> 
> [Seems easy enough](https://www.electron.build/file-patterns#multiple-glob-patterns).

We don't use electron-builder

**Review by RunDevelopment — APPROVED** [review]

**Pull request merged**